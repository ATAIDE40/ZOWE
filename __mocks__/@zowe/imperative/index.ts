/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Mocks all imperative classes specified under __mocks__/__imperative__
 * @returns mocked version of the @zowe/imperative module
 */
const __imperative__get = function () {
  const fs = require.requireActual("fs");
  const imperativePath = require("path").resolve(__dirname, "..", "..", "__imperative__");
  const extraClasses: any = {};
  fs.readdirSync(imperativePath).forEach((file: string) => {
    extraClasses[file.split(".")[0]] = require(`${imperativePath}/${file}`);
  });
  return {
    ...require.requireActual("@zowe/imperative"),
    // Note: Comment out the line below if you do not want to mock
    //       the classes in __mocks__/__imperative__/ for local testing purposes
    ...extraClasses
    // Note: You can still use Imperative.console.info("debug") in your test
  };
}

module.exports = __imperative__get();
