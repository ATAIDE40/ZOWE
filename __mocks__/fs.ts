/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

const __mocks__path = require("path");

let __mocks__fs = jest.genMockFromModule("fs") as any;
const __mocks__oldReadFileSync = require.requireActual("fs").readFileSync;

// This is a custom function that our tests can use during setup to specify
// what the files on the "mock" filesystem should look like when any of the
// `fs` APIs are used.
let __mocks__mockFiles = Object.create(null);
function __mocks__fs____setMockFiles(newMockFiles: { [key: string]: string }) {
    __mocks__mockFiles = Object.create(null);
    for (const file in newMockFiles) {
        const dir = __mocks__path.dirname(file);
        if (!__mocks__mockFiles[dir]) {
            __mocks__mockFiles[dir] = {};
        }
        __mocks__mockFiles[dir][__mocks__path.basename(file)] = newMockFiles[file];
    }
}

// A custom version of `readdirSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__readdirSync(filePath: string) {
    return __mocks__mockFiles[filePath] || {};
}

// A custom version of `existsSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__existsSync(filePath: string) {
    const fileContents = __mocks__fs__readdirSync(__mocks__path.dirname(filePath))[__mocks__path.basename(filePath)];
    if (typeof fileContents === "undefined") {
        return false;
    }
    return true;
}

// A custom version of `readFileSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__readFileSync(filePath: string, encoding?: string) {
    // Don't mock if yargs is trying to load a locale json file
    if (filePath.match(/node_modules.yargs/)) {
        return __mocks__oldReadFileSync(filePath, encoding);
    }

    if (!__mocks__fs__existsSync(filePath)) {
        throw new Error("File not found");
    }
    return __mocks__fs__readdirSync(__mocks__path.dirname(filePath))[__mocks__path.basename(filePath)];
}

// A custom version of `lstatSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__lstatSync(filePath: string) {
    return {
        isFile: () => false
    };
}

__mocks__fs.__setMockFiles = __mocks__fs____setMockFiles;
__mocks__fs.existsSync = __mocks__fs__existsSync;
__mocks__fs.readFileSync = __mocks__fs__readFileSync;
__mocks__fs.readdirSync = __mocks__fs__readdirSync;
__mocks__fs.lstatSync = __mocks__fs__lstatSync;

module.exports = __mocks__fs;
