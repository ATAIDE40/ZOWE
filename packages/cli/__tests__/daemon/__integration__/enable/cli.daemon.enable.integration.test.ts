/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ITestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { TestEnvironment } from "../../../../../../__tests__/__src__/environment/TestEnvironment";
import { ITestPropertiesSchema } from "../../../../../../__tests__/__src__/properties/ITestPropertiesSchema";
import * as fs from "fs";
import * as nodeJsPath from "path";

let testEnvironment: ITestEnvironment<ITestPropertiesSchema>;

describe("daemon enable", () => {
    const rimraf = require("rimraf").sync;
    let pathToBin: string;

    beforeAll(async () => {
        // Create the unique test environment
        testEnvironment = await TestEnvironment.setUp({
            testName: "daemon_enable_integration",
            skipProperties: true
        });
        pathToBin = nodeJsPath.normalize(testEnvironment.workingDir + "/bin");
    });

    beforeEach(async () => {
        // Remove any existing bin directory
        rimraf(pathToBin);
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(testEnvironment);
    });

    it("should display the help", async () => {
        const response = runCliScript(__dirname + "/__scripts__/daemon_enable_help.sh", testEnvironment);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });

    it("should fail if a bin file exists", async () => {
        fs.writeFileSync(pathToBin, "not a directory");
        const response = runCliScript(__dirname + "/__scripts__/daemon_enable.sh", testEnvironment);
        const stdoutStr = response.stdout.toString();
        expect(stdoutStr).toContain("Failed to enable daemon mode.");
        expect(stdoutStr).toContain(`The existing file '${pathToBin}' must be a directory.`);
        expect(response.status).toBe(1);
    });

    it("should place exe in a new bin dir", async () => {
        const response = runCliScript(__dirname + "/__scripts__/daemon_enable.sh", testEnvironment);
        expect(response.status).toBe(0);
        // todo: confirm exe exists
    });

    it("should place exe in an existing bin dir", async () => {
        fs.mkdirSync(pathToBin, 0o755);
        const response = runCliScript(__dirname + "/__scripts__/daemon_enable.sh", testEnvironment);
        expect(response.status).toBe(0);
        // todo: confirm exe exists
    });
});
