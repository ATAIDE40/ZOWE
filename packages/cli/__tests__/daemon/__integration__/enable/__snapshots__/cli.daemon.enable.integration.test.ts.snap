// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`daemon enable should display the help 1`] = `
"================ daemon enable help ===============

 COMMAND NAME
 ------------

   enable

 DESCRIPTION
 -----------

   Enables daemon-mode operation of the Zowe-CLI.

 USAGE
 -----

   zowe daemon enable [options]

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a group

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Enable daemon-mode:

      $ zowe daemon enable

"
`;
